### PR TITLE
Make wizard store dynamic registration data

### PR DESCRIPTION
**Pull request by TheOneRing**
So far, dynamic registration data (i.e., data the IdP uses to identify dynamically registered clients, the method of choice when, e.g., using Keycloak with oCIS) has not been stored by the setup wizard, causing users to be logged out and requiring them to log in again after the client restarts.

### CODE REVIEW

**Comment by ownclouders** [comment]
Results for <strong>GUI-tests</strong> https://drone.owncloud.com/owncloud/client/13080/6/1

:boom: The GUI tests failed.

GUI Logs: https://cache.owncloud.com/public/owncloud/client/13080/guiReportUpload/index.html

<details>

Server Logs: https://cache.owncloud.com/public/owncloud/client/13080/guiReportUpload/serverlog.log

Screenshots:
- [simple_sharing_of_folder_by_public_link_with_different_roles.png](https://cache.owncloud.com/public/owncloud/client/13080/guiReportUpload/screenshots/simple_sharing_of_folder_by_public_link_with_different_roles.png)

</details>

**Review by erikjv — APPROVED** [review]

**Review by michaelstingl — APPROVED** [review]
Behaves very well 👍

File: src/libsync/creds/credentialmanager.cpp [hunk 1]
```diff
@@ -211,7 +211,7 @@ void CredentialJob::start()
             _data = obj.toVariant();
             OC_ASSERT(_data.isValid());
         } else {
-            qCWarning(lcCredentaislManager) << "Failed to read client id" << _job->errorString();
+            qCWarning(lcCredentaislManager) << "Failed to get password" << _job->key() << _job->errorString();
```
Comment (TheOneRing) [inline src/libsync/creds/credentialmanager.cpp:214]: I guess we should use `scopedKey(_parent, _key)`

**Comment by TheOneRing** [comment]
LGTM: Please squash and take ownershipt of the change, as you did all the work :)

**Pull request merged**